tests/fixtures/envs/dummy/dummy_discrete_pixel_env.py

```diff
@@ -1,31 +1,60 @@
+from unittest import mock
+
 import gym
 import numpy as np
 
 from tests.fixtures.envs.dummy import DummyEnv
 
 
 class DummyDiscretePixelEnv(DummyEnv):
-    """A dummy discrete environment."""
+    """
+    A dummy discrete pixel environment.
+
+    It follows Atari game convention, where actions are 'NOOP', 'FIRE', ...
+    It also contains self.unwrapped.ale.lives, get_action_meanings for testing.
+    Observations are as following:
+        after reset    : np.zeros(self._shape).
+        action 1 (FIRE): np.ones(self._shape).
+        otherwise      : random if self.random is True,
+            otherwise previous state + action.
+    """
 
     def __init__(self, random=True):
         super().__init__(random)
+        self.unwrapped.get_action_meanings = self._get_action_meanings
+        self.unwrapped.ale = mock.Mock()
+        self.unwrapped.ale.lives = self.get_lives
+        self._shape = (10, 10, 3)
+        self._observation_space = gym.spaces.Box(
+            low=0, high=255, shape=self._shape, dtype=np.uint8)
+        self.step_called = 0
 
     @property
     def observation_space(self):
         """Return an observation space."""
-        self.shape = (10, 10, 3)
-        self._observation_space = gym.spaces.Box(
-            low=0, high=255, shape=self.shape, dtype=np.uint8)
         return self._observation_space
 
+    @observation_space.setter
+    def observation_space(self, observation_space):
+        self._observation_space = observation_space
+
     @property
     def action_space(self):
         """Return an action space."""
-        return gym.spaces.Discrete(2)
+        return gym.spaces.Discrete(5)
+
+    def _get_action_meanings(self):
+        return ['NOOP', 'FIRE', 'SLEEP', 'EAT', 'PLAY']
+
+    def get_lives(self):
```

Comment:
i don't see the point of this method -- it's only used once in the constructor and it's not part of the wrapper interface

Reply:
why is this a method?